### PR TITLE
[API][Web][Forms][SIMS-#1224] - Show PT courses on PIR completion

### PR DESCRIPTION
**Pull request by guru-aot**
When the student selects Part Time as their program intensity, adjust the existing course fields to allow the student to enter the following free text fields (add 100 character limit to free text fields):
Course Name
Course Code/Course Registration Number
Course Start/End Dates
-Fields are all or nothing complete
When Student submits, institution is able to view information below.
image

 When institution completes the PIR, show this course information with the institution on the PIR.
image

### CODE REVIEW

File: sources/packages/api/src/route-controllers/program-info-request/models/program-info-request.dto.ts [hunk 1]
```diff
@@ -32,10 +32,18 @@ export interface ProgramInfoRequestAPIOutDTO {
   offeringDelivered: string;
   offeringType: OfferingTypes;
   offeringIntensity: OfferingIntensity;
+  courseDetails?: CourseDetails[];
   pirDenyReasonId?: number;
   otherReasonDesc?: string;
 }
 
+export interface CourseDetails {
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/program-info-request/models/program-info-request.dto.ts:40]: Can we change it to a class and also the `ProgramInfoRequestAPIOutDTO`. I saw the other interfaces, but at least these ones are PR-related-ish.

File: sources/packages/web/package-lock.json [hunk 2]
```diff
@@ -30049,4 +30049,4 @@
       }
     }
   }
-}
+}
```
Comment (andrewsignori-aot) [inline sources/packages/web/package-lock.json:30052]: Not a real change to be part of the PR. Please avoid it next time 😉

File: sources/packages/forms/src/form-definitions/sfaa2021-22.json [hunk 3]
```diff
@@ -4421,348 +4421,1123 @@
...
+                          "size": "md",
+                          "currentWidth": 6
+                        },
+                        {
+                          "components": [
+                            {
+                              "label": "Course code",
+                              "tooltip": "Your course code is also known as your
course registration number typically shown
beside your course name (e.g. MGMT 101)",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/sfaa2021-22.json:4708]: It seems that there are some special characters in the middle of the string. Was it expected?
image

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work, just minor comments.

File: sources/packages/forms/src/form-definitions/programinformationrequest.json [hunk 4]
```diff
@@ -885,27 +277,1083 @@
...
+                                "labelMargin": "",
+                                "tag": "p",
+                                "className": "",
+                                "attrs": [
+                                  {
+                                    "attr": "",
+                                    "value": ""
+                                  }
+                                ],
+                                "content": "<Strong>Course name</Strong><br>\n{{ row.courseName }} {{ row.courseCode }}",
```
[the remark this answers is not in the document]
Comment (ann-aot) [reply sources/packages/forms/src/form-definitions/programinformationrequest.json:1091]: or `<p>`

File: sources/packages/forms/src/form-definitions/sfaa2021-22.json [hunk 5]
```diff
@@ -4421,348 +4421,1123 @@
...
+                              "customClass": "font-weight-bold",
```
Comment (ann-aot) [inline sources/packages/forms/src/form-definitions/sfaa2021-22.json:4709]: I think we have custom classes for form titles and labels

File: sources/packages/forms/src/form-definitions/programinformationrequest.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/programinformationrequest.json]: Used div 👍

File: sources/packages/forms/src/form-definitions/sfaa2021-22.json (hunk 3 above)
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/sfaa2021-22.json]: copied from figma, the words. Might have some special characters from them. Removed it now 👍

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, looks good 👍

**Pull request merged**